crates/common_enums/src/enums.rs

```diff
@@ -1305,6 +1305,19 @@ pub enum IntentStatus {
 
 impl IntentStatus {
     /// Indicates whether the syncing with the connector should be allowed or not
+    pub fn is_in_terminal_state(self) -> bool {
```

Comment:
Is the doc comment relevant here? I think this is supposed to be for the below function

Reply:
please add relevant comment here and the below function as well